Automattic/wp-calypso
# Components: Migrate to TypeScript

#### Changes proposed in this Pull Request

* Migrates as much of `@automattic/components` to TypeScript as possible

Leaves out Dialog because I had trouble figuring out how to decifer the button situation. I'll open a seperate PR for that as I think there will be some discussion around it.

#### Testing instructions



* Ensure type checks still pass (no functional changes to the components)

Fixes #
